main.py

```diff
@@ -35,107 +49,105 @@ def main():
     if args.channels != 1 and args.channels != 3:
         print('Invalid number of channels. Must be (1) for black/white or (3) for RGB')
 
-    # Text generation
     if args.script_file:
         with open(args.script_file, 'r') as f:
             script = f.read()
         print('script_file provided, skipping text generation')
-
     else:
         meditation_types = ['focused', 'body-scan', 'visualization', 'reflection', 'movement']
         if args.med_type not in meditation_types:
-            print('Invalid input. Provide valid med_type or input own script. Exitting...')
-            sys.exit()
-        script = text_generation(args.med_type)
-
-        script_file_name = f'{args.med_type}-meditation-script1.txt'
+            raise Exception('Invalid input. Provide valid med_type or input own script. Exitting...')
+
+        prompts = {
+            'focused': 'write me a focused meditation script designed to enhance focus and attention by noticing all 5 senses',
+            'body-scan': 'write me a body scan meditation script to relax and relieve stress by tightening and relaxing muscles',
+            'visualization': 'write me a visualization meditation script noticing all 5 senses at the beach/garden and is designed to boost mood, reduce stress, and promote inner peace',
+            'reflection': 'write me a reflection meditation script designed to increase self awareness, mindfulness, and gratitude by asking the user about the current day and the recent past',
+            'movement': 'write me a movement meditation script designed to improve mind body connection, energy, vitality, and the systems of the body'
+        }
+        prompt = prompts[args.med_type]
+
+        ###################
+        # Text generation #
+        ###################
+        script = text_generation(prompt)
+        script_file_name = f'data/{args.med_type}_meditation_script.txt'
         try:
             with open(script_file_name, 'w') as f:
                 f.write(script)
             print(f'Storing meditation script in {script_file_name}')
+        except ValueError as error:
+            print(f"Couldn't store script in file {script_file_name}: {error}")
+
+    ##################
+    # Text-to-speech #
+    ##################
+    if args.skip_audio_generation:
+        audio_filename = f"data/{args.med_type}_meditation_audio_{args.accent}.mp3"
+        text_to_speech(script, args.accent, audio_filename)
+
+        # Adding background music
+        if not os.path.isfile(args.music_file):
+            raise Exception('Music file not found')
+        audio_background_filename = f"data/{args.med_type}_meditation_audio_{args.accent}_background_music.mp3"
```

Comment:
Add "if not args.no_music:" below, so that we do not overlay the music if no_music is selected.

Reply:
Added `skip_background_music`.